lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -11,6 +11,15 @@ const Js::OpCode LowererMD::MDExtend32Opcode = Js::OpCode::MOVSXD;
 
 extern const IRType RegTypes[RegNumCount];
 
+bool
+LowererMDArch::IsArgSaveRequired(Func *func) {
+    return (func->IsJitInDebugMode() || func->IsGeneratorFunc() ||
+        func->GetJnFunction()->GetIsAsmjsMode() || func->IsLambda() ||
```

Comment:
Asmjs functions seem to be reading from rcx and rdx saves on stack. GlobalObject and InlineCache references are passed via these registers.